src/pint/utils.py

```diff
@@ -1686,6 +1694,103 @@ def remove_dummy_distance(c):
         return c
 
 
+def info_string(prefix_string="# ", comment=None):
+    """Returns an informative string about the current state of PINT.
+
+    Adds:
+    
+    * Creation date
+
+    * PINT version
+
+    * Username
+
+    * Host
+
+    * OS
+
+    * plus a user-supplied comment (if present).
+
+    Username is given by the ``gitpython`` global configuration ``user.name``
+    (if available), otherwise ``getpass.getuser()``.
+
+    Parameters
+    ----------
+    prefix_string: str, default='# '
+        a string to be prefixed to the output (often to designate as a
+        comment or similar)
+    comment: str, optional
+        a free-form comment string to be included if present
+    
+    Returns
+    -------
+    s : str
```

Comment:
No need to give it a name if there isn't an obvious one (e.g. Mc for a companion mass) - format will be okay with just type and description.